datahub/company_referral/test/test_views.py

```diff
@@ -11,13 +11,115 @@
 
 from datahub.company.test.factories import AdviserFactory, CompanyFactory, ContactFactory
 from datahub.company_referral.models import CompanyReferral
+from datahub.company_referral.test.factories import CompanyReferralFactory
 from datahub.core.test_utils import APITestMixin, create_test_user, format_date_or_datetime
 
 FROZEN_DATETIME = datetime(2020, 1, 24, 16, 26, 50, tzinfo=utc)
 
 collection_url = reverse('api-v4:company-referral:collection')
 
 
+class TestListCompanyListsView(APITestMixin):
+    """Tests for listing user's referrals."""
+
+    def test_returns_401_if_unauthenticated(self, api_client):
+        """Test that a 401 is returned if the user is unauthenticated."""
+        response = api_client.get(collection_url)
+        assert response.status_code == status.HTTP_401_UNAUTHORIZED
+
+    @pytest.mark.parametrize(
+        'permission_codenames,expected_status',
+        (
+            ([], status.HTTP_403_FORBIDDEN),
+            (['view_companyreferral'], status.HTTP_200_OK),
+        ),
+    )
+    def test_permission_checking(self, permission_codenames, expected_status, api_client):
+        """Test that the expected status is returned for various user permissions."""
+        user = create_test_user(permission_codenames=permission_codenames, dit_team=None)
+        api_client = self.create_api_client(user=user)
+        response = api_client.get(collection_url)
+        assert response.status_code == expected_status
+
+    def test_returns_empty_list_when_no_lists(self):
+        """Test that no results are returned when the user has no referrals."""
+        response = self.api_client.get(collection_url)
+
+        assert response.status_code == status.HTTP_200_OK
+        response_data = response.json()
+        assert response_data['results'] == []
+
+    def test_doesnt_return_other_users_referrals(self):
+        """Test that other users' referrals are not returned."""
+        # Create some referrals sent or received by other users
+        CompanyReferralFactory.create_batch(5)
```

Comment:
I think 1 or 2 objects should be enough (as 2 should not be any different from 5).
```suggestion
        CompanyReferralFactory.create_batch(2)
```

Reply:
Good point! Yes, having just two objects will be sufficient. Thank you!